OpenRA.Mods.Common/TraitsInterfaces.cs

```diff
@@ -162,7 +162,8 @@ public interface INotifyChat { bool OnChat(string from, string message); }
 	public interface INotifyProduction { void UnitProduced(Actor self, Actor other, CPos exit); }
 	public interface INotifyOtherProduction { void UnitProducedByOther(Actor self, Actor producer, Actor produced, string productionType, TypeDictionary init); }
 	public interface INotifyDelivery { void IncomingDelivery(Actor self); void Delivered(Actor self); }
-	public interface INotifyDocking { void Docked(Actor self, Actor harvester); void Undocked(Actor self, Actor harvester); }
+	public interface INotifyDock { void Docked(Actor self, Actor dockable); void Undocked(Actor self, Actor dockable); }
```

Comment:
The name `INotifyDock` is pretty obscure. It took most of the context of this commit to understand what's behind it.

Reply:
The idea is that

Dock - an actor to which a unit docks to
Dockable - an actor which docks to another actor

In in essence INotifyDock notifies the actor which has a dock

Reply:
How about `INotifyDocked`?

Reply:
what would be the rationale behind that? 

I'm planning to have `Dock` trait and `DockableBase` abstract class

Reply:
I'm not going to comment on the future names or purposes at this time, but tldr we need 2 names to signify the two sides here. Unfortunately I can't come up with a better name for `INotifyDock` right now other than changing it to `IDock` (which also isn't great; hence my proposal of `AllowsDocking`, which is pretty clear about its purpose).

Reply:
the most important part is for it to be clear which interface should be used, so naming after the type of actor it should be used on makes the most sense. Also see the opening description

Reply:
If you really want to be explicit you could name it something along the lines of `INotifyDock(ing)Host`.

Reply:
renamed to INotifyDockingClient, INotifyDockingHost

Reply:
You named them `INotifyDockClient/Host`, just for the record.